### PR TITLE
Add conversion script from PLY to NDT

### LINKED ISSUE #348: Add conversion script from `pbstream` to NDT map
### Feature description

Basically what the title says.
This'll unblock #316 .

### PR DESCRIPTION
**Pull request by nahueespinosa**
### Proposed changes

Closes #348.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

- [x] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: beluga_ros/src/beluga_ros/ply_to_ndt.py
Comment (nahueespinosa) [inline beluga_ros/src/beluga_ros/ply_to_ndt.py:1]: This has the same structure as `occupancy_grid_to_ndt.py` for consistency.

File: beluga_ros/src/beluga_ros/ply_reader.py
Comment (nahueespinosa) [inline beluga_ros/src/beluga_ros/ply_reader.py:1]: Will add some tests for this before merging.

File: beluga_ros/src/beluga_ros/ply_reader.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+
+def read_ply_header(file: BinaryIO):
+    """Read and validate the contents of the header of a PLY file."""
+    assert file.readline().rstrip() == b'ply', 'Missing ply at the start of the file'
+
+    keyword, format, version = file.readline().decode().split()
+
+    assert keyword == "format", 'Missing format keyword'
+    assert format in ['ascii', 'binary_little_endian', 'binary_big_endian']
```
Comment (serraramiro1) [inline beluga_ros/src/beluga_ros/ply_reader.py:68]: Super nit: Use tuples for immutable data

File: beluga_ros/src/beluga_ros/ply_reader.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+            break
+
+        keyword, *values = line.decode().split()
+
+        if keyword == 'comment':
+            continue  # Ignore comments
+
+        if keyword == 'element':
+            name, count = values
+            assert name in ['vertex', 'face', 'edge']
```
Comment (serraramiro1) [inline beluga_ros/src/beluga_ros/ply_reader.py:85]: Same here

File: beluga_ros/src/beluga_ros/ply_to_ndt.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        type=Path,
+        help="Path to the directory where outputs will be dumped to. "
+        "Will be created if it doesn't exist.",
+        required=True,
+    )
+    parser.add_argument(
+        '-c',
+        '--cell_size',
+        type=float,
+        default=0.1,
```
Comment (serraramiro1) [inline beluga_ros/src/beluga_ros/ply_to_ndt.py:46]: I love that the default cell size is way smaller :)

**Comment by serraramiro1** [comment]
Looks great! 

Meta question:
Did you consider using a third party python module for loading in the plys?

I feel like even though this looks sane, we won't cover all the combinations in the tests and might aswell rely on a trusted module.
Something like https://github.com/dranjan/python-plyfile comes to mind.

Not blocking at all

**Comment by hidmic** [comment]
> Did you consider using a third party python module for loading in the plys?

+1000 to this.

**Comment by nahueespinosa** [comment]
> Did you consider using a third party python module for loading in the plys?

Where is the fun in that? :disappointed: 

Haha, comments addressed, I think.

**Review by serraramiro1 — APPROVED** [review]
LGTM!

**Pull request merged**